src/java.security.jgss/share/classes/sun/security/jgss/wrapper/GSSCredElement.java

```diff
@@ -69,6 +70,8 @@ void doServicePermCheck() throws GSSException {
         cStub = GSSLibStub.getInstance(mech);
         usage = GSSCredential.INITIATE_ONLY;
         name = srcName;
+
+        Cleaner.create().register(this, this::dispose);
```

Comment:
This will create a memory leak: `this` will never be garbage collected if it's referenced by the cleaner action.

Reply:
The cleaner gets triggered when the instance is unreachable, no values from that instance can be modified or referenced.
Any information needed for the cleanup must be in a separate object.
In this case, that is pCred and cStub. 
Typically, new nested class is defined that holds the pCred and cStub and has a run() method that 
calls `cStub.releaseCred(pCred)`.
But the package access to non-final `pCred` raises a flag about when its mutated.
If its mutated after the cleaner is created, the cleaner will release the wrong cred.

If the GSSCredElement is itself just a holder of the credentials, then perhaps the cleaner should be triggered on the instance that is referring to the GSSCredElement being unreachable.

Reply:
Yes.  I'm working on an update, and more code changes (including immutable objects) may be placed.  Thank you @dfuch and @RogerRiggs for the good catch.  I totally forgot about the object reference problems while using Cleaner.

Reply:
Hmm, the earlier JCE change would also needs to be updated as it calls a cleanup method on the to-be-cleaned object.

Reply:
> Hmm, the earlier JCE change would also needs to be updated as it calls a cleanup method on the to-be-cleaned object.

Yes, I will check the cleaner used in the security components and make sure there is object reference problems.

Reply:
> I'm not sure if it's possible to write a test on the cleanup, but at least we can temporarily add a `println` line there and see if it ever gets called.

I did not find a way to test cleanup yet.  Yes, a temporary 'println" is what I can use for now.